228: Implement ability to manually sort the glossary facet for the dictionary

m4olivei opened the pull request:
Resolves #228 

### Description

Implements the ability to control the dictionary glossary order.

### Testing Instructions

- [ ] Add some Dictionary Word nodes with and without customized Glossary Entry values
- [ ] Go to `/dictionary`. Should see default unicode ordering for the glossary entries
- [ ] Go to `/admin/config/mukurtu/dictionary-glossary-order`
- [ ] Set the sort to "User-defined sort"
- [ ] Manually set weights for known glossary entry values
- [ ] Go to `/dictionary`. Should see your configured weights reflected in the display of the glossary entries

NOTE: The default behavior of facets is to only show values that exist in the system. If we want to give admins the option to flesh out entries to fill the alphabet with empty links, let’s do that as a follow-up. It feels like that would just present noise in the UX, and we're better off not having that option.

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr1303-iq6mw1mq1ncsi9oxnt1nmo6zk9rcm1yl.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6953f393e84840977f54bc54

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr1303-iq6mw1mq1ncsi9oxnt1nmo6zk9rcm1yl.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6953f393e84840977f54bc54

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr1303-iq6mw1mq1ncsi9oxnt1nmo6zk9rcm1yl.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6953f393e84840977f54bc54

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr1303-iq6mw1mq1ncsi9oxnt1nmo6zk9rcm1yl.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6953f393e84840977f54bc54

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr1303-iq6mw1mq1ncsi9oxnt1nmo6zk9rcm1yl.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6953f393e84840977f54bc54

m4olivei commented:
Gah, hang on, not there just yet. While the facets are showing distinctly now, the filtering is not correct yet :/

m4olivei commented:
@nick-deer @michael-wynne-wsu This is ready for re-review. As I noted in my last comment, we're somewhat restrained here in that the facets will now be very verbose by default, and will expose all of the following if there are `field_glossary_entry` values for each one:

* e
* E
* é
* É

This can be mitigated if desired by the user, by editors adopting a convention of using either upper or lower case consistently if they want exposed facets to collapse down, for instance if they only want uppercase to show in the facets, they would set `field_glossary_entry` to either "E" or "É", where the term itself can be lowercase. Although it is perhpas desireable to have that level of verbosity, if for instance an editorial team adopts a style guide that reserves uppercase for proper nouns, and they actually want all 4 facets. Let me know what you think.

m4olivei commented:
> @m4olivei That's fantastic. I see the potential issues with that, but I think we can work with it just as you described. I suspect it will be rare for it to cause any difficulty.
> 
> And I see the facet filtering is also working, great!
> 
> Hopefully one last note... Regardless of user/default glossary sort, the results are in a different order from that setting, and I would expect that to match. I actually can't really tell what the order of results are keyed to.
> 
> <img alt="Screenshot 2026-01-08 at 1 25 21 PM" width="1404" height="622" src="https://private-user-images.githubusercontent.com/88502274/533591603-d180d540-d79a-4522-89fc-3a203aaea231.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc5MzEwMjEsIm5iZiI6MTc2NzkzMDcyMSwicGF0aCI6Ii84ODUwMjI3NC81MzM1OTE2MDMtZDE4MGQ1NDAtZDc5YS00NTIyLTg5ZmMtM2EyMDNhYWVhMjMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTA5VDAzNTIwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYyZWYwNWUyYTA4NWNlOGM5YmZlOWVjZTUxNDIxNzA1MTVmYWVlNDhkNjFmODU3NzRhZWNiMzk5M2VjMjJmNWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tREfDI-cXY3sLzAfSsIY1jLHKUcFX-hpy8_z3eAUbC0">

Thanks @michael-wynne-wsu. I've updated sorting to match the manually set Dictionary Glossary Order when applicable. That was tricky to do given the configuration is removed from the actual data and we need it to be part of the views sort, which is a Search API based view. AI was a huge help though, and I think we came up with a good solution for that here. 

I also added a "Reset Order to Alphabetical" action on the "Dictionary Glossary Order" form (`/admin/config/mukurtu/dictionary-glossary-order`). This can be useful if a user has set a manual order for things, but then later added a number of words which end up floating to the end of the previously manually sorted glossary entries. This lets a user start from a basis of alphabetical again, and re-arrange from there. This might be only useful for testing scenarios like we find ourselves in now, but maybe it will be more broadly useful:

<img width="3424" height="1506" alt="image" src="https://github.com/user-attachments/assets/b7cd847f-f7fb-48ac-a4f4-a5bdcf8dde63" />
